server/controllers/assets/utils/determineRequestType.js

```diff
@@ -0,0 +1,61 @@
+const logger = require('winston');
+const { EMBED, BROWSER, SOCIAL } = require('../constants/request_types.js');
+
+function headersMatchesSocialBotList (headers) {
+  const socialBotList = [
+    'facebookexternalhit',
+    'Twitterbot',
+  ];
+  const userAgent = headers['user-agent'];
+  for (let i = 0; i < socialBotList.length; i++) {
+    const socialBot = socialBotList[i];
+    if (userAgent.indexOf(socialBot) >= 0) {
+      logger.debug('headers on request matched this bot:', socialBot);
+      return true;
+    }
+  }
+  return false;
+}
+
+function clientAcceptsHtml ({accept}) {
+  return accept && accept.match(/text\/html/);
+};
+
+function requestIsFromBrowser (headers) {
+  return headers['user-agent'] && headers['user-agent'].match(/Mozilla/);
+};
+
+function clientWantsAsset ({accept, range}) {
+  const imageIsWanted = accept && accept.match(/image\/.*/) && !accept.match(/text\/html/) && !accept.match(/text\/\*/);
+  const videoIsWanted = accept && range;
+  return imageIsWanted || videoIsWanted;
+};
+
+const determineRequestType = (hasFileExtension, headers) => {
+  let responseType;
+  logger.debug('headers:', headers);
+  // return early with 'show' if headers match the list
+  if (headersMatchesSocialBotList(headers)) {
+    return SOCIAL;
+  }
+  // if request is not from a social bot...
+  if (hasFileExtension) {
```

Comment:
Multiple return statements in a function can be a contentious topic (e.g. https://stackoverflow.com/questions/36707/should-a-function-have-only-one-return-statement), but I think a function like this is better if the returns are just inlined, rather than setting a variable.